Tasks/Common/PowershellHelpers/Helpers.ps1

```diff
@@ -37,7 +38,8 @@ function Invoke-ActionWithRetries
 
         try
         {
-            $result = & $Action
+            $result = & $Action $lastResult
```

Comment:
Have you thought about not passing $lastResult and instead caller handling this in $ActionSuccessValidator?

Reply:
Will you remove $lastResult in this PR itself?

Reply:
We can do that in a seperate PR.

Reply:
Better to do in this PR itself - before it gets started being used by other callers